### PR TITLE
Applying sort-drill thru should REPLACE existing sort on same column

### LINKED ISSUE #34497: [MLv2] [BE] Applying `sort` to already sorted column doesn't remove initial sorting
Applying `sort` to already sorted column doesn't remove initial sorting, but always adds another one.

https://github.com/metabase/metabase/assets/7983744/0d1e0a97-0814-4813-a99e-9871764ca006

You can use this reproduction unit test - 
https://github.com/metabase/metabase/blob/fc358ef7c24ada415997ea781b3b1d706439a036/frontend/src/metabase-lib/drills.unit.spec.ts#L1593-L1613

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #34497

### CODE REVIEW

File: src/metabase/lib/drill_thru/sort.cljc [hunk 1]
```diff
@@ -46,8 +61,13 @@
   ([query                        :- ::lib.schema/query
     stage-number                 :- :int
     {:keys [column], :as _drill} :- ::lib.schema.drill-thru/drill-thru.sort
-    direction                    :- [:enum :asc :desc]]
-   (lib.order-by/order-by query stage-number column (keyword direction))))
+    direction                    :- ::lib.schema.order-by/direction]
```
Comment (metamben) [inline src/metabase/lib/drill_thru/sort.cljc:64]: 👍

File: src/metabase/lib/drill_thru/sort.cljc [hunk 2]
```diff
@@ -46,8 +61,13 @@
...
+   ;; if you have an existing order by, the drill thru returned by [[sort-drill]] would only be one that would suggest
+   ;; changing it to the opposite direction, so we can safely assume we want to change the direction and
+   ;; use [[lib.order-by/change-direction]] here.
```
Comment (metamben) [inline src/metabase/lib/drill_thru/sort.cljc:67]: I still find it counterintuitive to set `:asc` if the drill says `:sort-directions [:desc]` or if `direction` is `:desc`. Would it be difficult or strange to set `direction`?

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/lib/drill_thru/sort.cljc (hunk 2 above)
Comment (camsaul) [reply src/metabase/lib/drill_thru/sort.cljc:67]: Yeah, I agree with you and I was going to push a change for this but I guess it auto-merged first. Sorry, I'll fix it in a follow on PR

**Pull request merged**